dev/NavigationView/NavigationViewItem.cpp

```diff
@@ -158,6 +173,48 @@ void NavigationViewItem::OnPropertyChanged(const winrt::DependencyPropertyChange
     {
         UpdateVisualStateNoTransition();
     }
+
+    if (auto node = TreeNode())
+    {
+        if (property == s_IsExpandedProperty)
+        {
+            //TODO: Instead of toggling, read specific bool value
+            ToggleIsExpanded(node);
+            
+            UpdateSelectionIndicatorVisiblity();
+        }
+        else if (property == s_MenuItemsSourceProperty)
+        {
+            winrt::IInspectable value = args.NewValue();
+            winrt::get_self<TreeViewNode>(node)->ItemsSource(value);
+            //m_dispatcherHelper.RunAsync(
+            //    [node, value]()
+            //{
+            //    winrt::get_self<TreeViewNode>(node)->ItemsSource(value);
+            //});
+        }
+        else if (property == s_IsChildSelectedProperty)
+        {
+            // Update the corresponding node
+            if (auto navViewList = GetNavigationViewList())
+            {
+                auto viewModel = winrt::get_self<NavigationViewList>(navViewList)->ListViewModel();
+                TreeNodeSelectionState selectionState = IsChildSelected() ? TreeNodeSelectionState::PartialSelected : TreeNodeSelectionState::UnSelected;
```

Comment:
IsChildSelected is writable for the scenarios where a developer would potentially want to select a MenuItem that has not been realized yet.
I never actually set a child node to selected because I use the treeview selection logic only for the purposes of determining the parents of a selected item (the behavior of changing the state of all ancestors was already in place, so seemed like a good idea to utilize it).